src/app/modules/core/auth/models/user.ts

```diff
@@ -0,0 +1,101 @@
+import { Role } from 'src/app/shared/models/role.enum'
+
+export interface IName {
+  first: string
+  middle?: string
+  last: string
+}
+
+export enum PhoneType {
+  None = 'none',
+  Mobile = 'mobile',
+  Home = 'home',
+  Work = 'work',
+}
+
+export interface IPhone {
+  type: PhoneType
+  digits: string
+  id: number
+}
+
+export interface IUser {
+  _id: string
+  email: string
+  name: IName
+  readonly fullName?: string
+  picture: string
+  role: Role | string
+  userStatus: boolean
+  dateOfBirth: Date | null | string
+  level: number
+  address: {
+    line1: string
+    line2?: string
+    city: string
+    state: string
+    zip: string
+  }
+  phones: IPhone[]
+}
+
+export class User implements IUser {
+  constructor(
+    // tslint:disable-next-line:variable-name
+    public _id = '',
+    public email = '',
+    public name = { first: '', middle: '', last: '' } as IName,
+    public picture = '',
+    public role = Role.None,
+    public dateOfBirth: Date | null = null,
+    public userStatus = false,
+    public level = 0,
+    public address = {
+      line1: '',
+      city: '',
+      state: '',
+      zip: '',
+    },
```

Comment:
For the address you can use the interface, as done with the entity models, so that here you also specify what type it is, but now that's okay too